nomad/api/variables.py

```diff
@@ -0,0 +1,70 @@
+from nomad.api.base import Requester
+
+
+class Variables(Requester):
+
+    """
+    The /vars endpoints are used to query for and interact with variables.
+    https://developer.hashicorp.com/nomad/api-docs/variables
+    """
+
+    ENDPOINT = "vars"
+
+    def __init__(self, **kwargs):
+        super(Variables, self).__init__(**kwargs)
+
+    def __str__(self):
+        return "{0}".format(self.__dict__)
+
+    def __repr__(self):
+        return "{0}".format(self.__dict__)
+
+    def __getattr__(self, item):
+        raise AttributeError
+
+    def __contains__(self, item):
+        try:
+            variables = self.get_variables()
+
+            for var in variables:
+                if var["ID"] == item:
+                    return True
+            else:
+                return False
+        except nomad.api.exceptions.URLNotFoundNomadException:
+            return False
+
+    def __getitem__(self, item):
```

Comment:
Would it not be better to use the `get_variable` API for this?

It also seems like, though convient, `__getitem__` and `__contains__` are expensive (making an API call and iterating through all variables) for every check. It also doesn't add value over using the `Variable` interface.

There is some risk in making it convenient or too easy to do something expensive as it makes it more likely users will do the wrong thing. It might be worth considering either adding a cache or dropping these features in favor of `Variable`. Or at least using `Variable` under the hood to avoid the loop here.

Reply:
In reading through the rest of the library, it seems like this is a common pattern, so probably best to leave it as is for consistency baring a proposal to shift patterns.

Reply:
It was just a draft :)

I changed on `get_variables` now.

Reply:
Hah. Sorry for jumping the gun. I got excited.

Reply:
NP :)
Thanks for your help!